go/tools/builders/compilepkg.go

```diff
@@ -461,3 +469,15 @@ func sanitizePathForIdentifier(path string) string {
 		return '_'
 	}, path)
 }
+
+func extractPkgDef(fromArchive, toFile string) error {
```

Comment:
I think `readFileInArchive` should accept a file name instead of a predicate, so this function could be dropped, and callers could do something like `readFileInArchive("foo.a", "__.PKGDEF")`.